src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -128,21 +118,71 @@ static string GetActionsListForLogging(IReadOnlyList<P.OrchestratorAction> actio
             }
         }
 
+        async ValueTask<OrchestrationRuntimeState> BuildRuntimeStateAsync(
+            P.OrchestratorRequest orchestratorRequest,
+            CancellationToken cancellation)
+        {
+            IEnumerable<HistoryEvent> pastEvents = [];
+            if (orchestratorRequest.RequiresHistoryStreaming)
```

Comment:
consider moving this block into its own method `GetAllHistoryAsync(..)` or similarly named.

Reply:
It's not that much code and isn't required elsewhere, so I'd prefer to keep it inlined to minimize the code's overall conceptual overhead.